modules/api/src/main/java/org/apache/fluo/api/observer/ObserversFactory.java

```diff
@@ -0,0 +1,119 @@
+ * the License.
+ */
+
+package org.apache.fluo.api.observer;
+
+import java.util.function.BiConsumer;
+
+import org.apache.fluo.api.client.FluoClient;
+import org.apache.fluo.api.config.FluoConfiguration;
+import org.apache.fluo.api.config.SimpleConfiguration;
+import org.apache.fluo.api.data.Bytes;
+import org.apache.fluo.api.data.Column;
+import org.apache.fluo.api.metrics.MetricsReporter;
+import org.apache.fluo.api.observer.Observer.NotificationType;
+
+/**
+ * Fluo Workers use this class to create {@link Observer}s to process notifications. Implementations
+ * of this class should emit zero or more {@link Observer}s.
+ *
+ * <p>
+ * When Fluo is initialized {@link #getObservedColumns(Context, BiConsumer)} is called. The columns
+ * it emits are stored in Zookeeper. Transactions will use the columns stored in Zookeeper to
+ * determine when to set notifications. When Workers call
+ * {@link #createObservers(ObserverConsumer, Context)}, the columns emitted must be the same as
+ * those emitted during initialization. If this is not the case, then the worker will fail to start.
+ *
+ * @see FluoConfiguration#setObserversFactory(String)
+ * @since 1.1.0
+ */
+public interface ObserversFactory {
+
+  /**
+   * @since 1.1.0
+   */
+  interface Context {
+    /**
+     * @return A configuration object with application configuration like that returned by
+     *         {@link FluoClient#getAppConfiguration()}
+     */
+    SimpleConfiguration getAppConfiguration();
+
+    /**
+     * @return A {@link MetricsReporter} to report application metrics from observers.
+     */
+    MetricsReporter getMetricsReporter();
+  }
+
+  /**
+   * Observers are emitted to the worker using this interface. This interface also allows
+   * {@link Observer}s to be related to the columns that trigger them.
+   *
+   * @since 1.1.0
+   */
+  interface ObserverConsumer {
+    void accept(Column observedColumn, NotificationType ntfyType, Observer observer);
+
+    /**
+     * This method was created to allow Observers written as lambda to be passed {@link String}
+     * instead of {@link Bytes}.
+     * 
+     * <pre>
+     * <code>
+     *   void createObservers(ObserverConsumer obsConsumer, Context ctx) {
+     *     obsConsumer.accepts(someColumn, WEAK, (tx,row,col) -> {
+     *      //row is of type String
+     *     };
+     *   }
+     * </code>
+     * </pre>
+     */
+    void accepts(Column observedColumn, NotificationType ntfyType, StringObserver observer);
+  }
+
+  /**
+   * This is method is called by Fluo Workers to create observers to process notifications.
+   *
+   * <p>
+   * Observers emitted may be called concurrently by multiple threads to process different
+   * notifications. Observers should be tolerant of this.
+   *
+   * @param obsConsumer Emit an applications observers to this consumer.
+   */
+  void createObservers(ObserverConsumer obsConsumer, Context ctx);
```

Comment:
> Why not make this interface a `Function<Context,Stream<Observer>>` 

Need to relate the observer to a column, so can't just return a stream of Observers.

> or maybe make the factory return a set of observers?

As for returning a set, I initially had something like that.  But I changed it because if forced longer, boiler plate code.

For example, below is the case where the user has to return a Map.

```java
 Map<ObservedColumn, Observer> createObservers(Context ctx) {
   Map<ObservedColumn, Observer> observers = new HashMap<>(); //boilerplate code
   observers.put(new ObservedColumn(col1,NotificationType.STRONG), (tx,row,col)->{...});  //this is longer than the observer consumer code
   observers.put(new ObservedColumn(col2,NotificationType.STRONG), (tx,row,col)->{...});
   return observers; //boilerplate code
}
```

Below is the same code with the observer consumer.   Its shorter and lacks the boilerplate code.

```java
 Map<ObservedColumn, Observer> createObservers(ObserverConsumer oc, Context ctx) {
   oc.accept(col1,NotificationType.STRONG), (tx,row,col)->{...});  
   oc.accept(col2,NotificationType.STRONG), (tx,row,col)->{...});
}
```

Another nice thing I found with the ObserverConsumer, is its a standard thing that can be passed to other code.  For example it can be passed to Fluo Recipes code that creates Observers.